apps/core/tests/test_mutations.py

```diff
@@ -356,3 +357,64 @@ def test_table_join(self, table_join_delay_func):
         content = resp_data["data"]["tableJoin"]
         assert content["ok"] is True
         table_join_delay_func.assert_called_with(int(content["result"]["id"]))
+
+    def test_table_join_preview(self):
+        # Set preview data
+        preview_mutation = """
+            mutation Mutation($tableId: ID! $data: TableJoinInputType!) {
+                joinPreview(id: $tableId, data: $data) {
+                    ok
+                    errors
+                    result {
+                        rows
+                        columns {
+                            key
+                            label
+                        }
+                    }
+                }
+            }
+        """
+        src_cols, _, src_rows = get_dummy_source_data()
+        tgt_cols, _, tgt_rows = get_dummy_target_data()
+        self.source_table.preview_data = {
+            "columns": src_cols,
+            "rows": src_rows,
+        }
+        self.target_table.preview_data = {
+            "columns": tgt_cols,
+            "rows": tgt_rows,
+        }
+        self.source_table.save()
+        self.target_table.save()
+
+        variables = {
+            "tableId": self.source_table.id,
+            "data": {
+                "targetTable": self.target_table.id,
+                "joinType": self.genum(Join.JoinType.INNER_JOIN),
+                "clauses": json.dumps(
+                    [
+                        {
+                            "source_column": "id",
+                            "target_column": "id",
+                            "operation": JOIN_CLAUSE_OPERATIONS.EQUAL,
+                        }
+                    ]
+                ),
+            },
+        }
+
+        resp_data = self.query_check(
+            preview_mutation,
+            variables=variables,
+        )
+        data = resp_data["data"]["joinPreview"]["result"]
+        _test_expectation(
```

Comment:
@bewakes Any idea if we can test if new columns are appended after joining?

Reply:
It has been included.